modules/social_features/social_event/src/Plugin/GraphQL/SchemaExtension/EventSchemaExtension.php

```diff
@@ -0,0 +1,160 @@
+<?php
+
+namespace Drupal\social_event\Plugin\GraphQL\SchemaExtension;
+
+use Drupal\graphql\GraphQL\ResolverBuilder;
+use Drupal\graphql\GraphQL\ResolverRegistryInterface;
+use Drupal\graphql\Plugin\GraphQL\SchemaExtension\SdlSchemaExtensionPluginBase;
+
+/**
+ * Adds event data to the Open Social GraphQL API.
+ *
+ * @SchemaExtension(
+ *   id = "social_event_schema_extension",
+ *   name = "Open Social - Event Schema Extension",
+ *   description = "GraphQL schema extension for Open Social event data.",
+ *   schema = "open_social"
+ * )
+ */
+class EventSchemaExtension extends SdlSchemaExtensionPluginBase {
+
+  /**
+   * {@inheritdoc}
+   */
+  public function registerResolvers(ResolverRegistryInterface $registry) {
+    $builder = new ResolverBuilder();
+
+    $this->addQueryFields($registry, $builder);
+    $this->addEventFields($registry, $builder);
+  }
+
+  /**
+   * Registers type and field resolvers in the shared registry.
+   *
+   * @param \Drupal\graphql\GraphQL\ResolverRegistryInterface $registry
+   *   The resolver registry.
+   * @param \Drupal\graphql\GraphQL\ResolverBuilder $builder
+   *   The resolver builder.
+   */
+  protected function addEventFields(ResolverRegistryInterface $registry, ResolverBuilder $builder) {
+    $registry->addFieldResolver('Event', 'id',
+      $builder->produce('entity_uuid')
+        ->map('entity', $builder->fromParent())
+    );
+
+    $registry->addFieldResolver('Event', 'title',
+      $builder->compose(
+        $builder->produce('entity_label')
+          ->map('entity', $builder->fromParent()),
+        $builder->produce('uppercase')
+          ->map('string', $builder->fromParent())
+      )
+    );
+
+    $registry->addFieldResolver('Event', 'author',
+      $builder->produce('entity_owner')
+        ->map('entity', $builder->fromParent())
+    );
+
+    $registry->addFieldResolver('Event', 'bodyHtml',
+      $builder->fromPath('entity:node', 'body.0.value')
```

Comment:
This needs rendering similarly to https://github.com/goalgorilla/open_social/blob/55e5b28c2750bd9d4a5da1b18c4c4836fed67a65/modules/social_features/social_comment/src/Plugin/GraphQL/SchemaExtension/CommentSchemaExtension.php#L50 as the value may contain invalid Drupal specific HTML.

Reply:
Changed.